blobfuse/blobfuse.cpp

```diff
@@ -535,12 +546,60 @@ int set_log_mask(const char * min_log_level_char)
         return 0;
     }
 
-    syslog(LOG_CRIT, "Unable to start blobfuse. Error: Invalid log level \"%s\"", min_log_level.c_str());
-    fprintf(stderr, "Error: Invalid log level \"%s\".  Permitted values are LOG_OFF, LOG_CRIT, LOG_ERR, LOG_WARNING, LOG_INFO, LOG_DEBUG.\n", min_log_level.c_str());
-    fprintf(stdout, "If not specified, logging will default to LOG_WARNING.\n\n");
+    if (blobfuseInit) {
+        syslog(LOG_CRIT, "Unable to start blobfuse. Error: Invalid log level \"%s\"", min_log_level.c_str());
+        fprintf(stderr, "Error: Invalid log level \"%s\".  Permitted values are LOG_OFF, LOG_CRIT, LOG_ERR, LOG_WARNING, LOG_INFO, LOG_DEBUG.\n", min_log_level.c_str());
+        fprintf(stdout, "If not specified, logging will default to LOG_WARNING.\n\n");
+    } else {
+        set_log_mask(options.log_level, false);
+    }
     return 1;
 }
 
+int refresh_from_config_file(const std::string configFile)
+{
+    std::ifstream file(configFile);
+    if(!file)
+    {
+        syslog(LOG_CRIT, "Unable to read config file : %s", configFile.c_str());
+        return -1;
+    }
+
+    std::string line;
+    bool logLevelFound = false;
+    while(std::getline(file, line))
+    {
+        // skip over comments
+        if(line[0] == '#') {
+            continue;
+        }
+
+       std::size_t pos = line.find("logLevel");
+        if(pos != std::string::npos)
+        {
+           std::string logLevel = line.substr(line.find(" ")+1);
+           str_options.logLevel = trim(logLevel);
+            logLevelFound = true;
+        }
+    }
+
+    if(!logLevelFound) {
+        str_options.logLevel = (options.log_level) ? : "";
```

Comment:
We should not reset the original loglevel what if it was specified in the commandline. If log_level is not found nothing should happen.
Remove this if condition.

Reply:
This condition means if logLevel was not found in the config file while processing the signal then we go back to the logging level supplied in the command-line.  str_options represent the config being read from file while options is the global variable representing command line options.